packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -270,14 +271,13 @@ export class DataStores implements IDisposable {
     }
 
     public async getDataStore(id: string, wait: boolean): Promise<IFluidDataStoreChannel> {
-        const deferredContext = this.contexts.prepDeferredContext(id);
+        const contextP = this.contexts.getBoundOrRemoted(id, wait);
```

Comment:
why not just await here, rather than hanging on to the promise

Reply:
Well I did before but then it looked weird, as if we're ignoring the `wait` param. See the last commit. Happy to revert if you like it better before, I was on the fence.

Reply:
the method is always async, regardless of the wait param, so the caller will always need to await. i'm not sure i see what you are trying to do.